### PR TITLE
GitHub oauth

### PR DESCRIPTION
**Pull request by KirbyDD**
This PR is for having a user connect their github account with their current account on our site.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -41,6 +43,7 @@ group :development, :test do
   gem 'simplecov'
   gem 'selenium-webdriver'
   gem 'webdrivers'
+  gem 'chromedriver-helper'
```
Comment (zachholcomb) [inline Gemfile:46]: oh that's where the chromedriver goes in rails, makes sense

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,10 @@ rbenv:
 addons:
   postgresql: 9.6
   chrome: stable
-dist: trusty
+dist: xenial
+before_install:
+  - yes | gem update --system --force
+  - gem install bundler
```
Comment (zachholcomb) [inline .travis.yml:10]: makes sense you have to install the gem for the driver we required cool

File: app/assets/stylesheets/application.sass [hunk 3]
```diff
@@ -4,3 +4,4 @@
 @import "buttons"
 @import "sortable"
 @import "admin"
+@import "dashboard"
```
Comment (zachholcomb) [inline app/assets/stylesheets/application.sass:7]: so is this the github dashboard css

File: config/routes.rb [hunk 4]
```diff
@@ -24,6 +24,9 @@
     end
   end
 
+  get '/auth/github'
+  get '/auth/github/callback', to: "users#update"
```
Comment (zachholcomb) [inline config/routes.rb:28]: so that's where the api response is being routed to in our app

**Comment by zachholcomb** [comment]
looks good, have a couple questions about what is happening but overall it's starting to make more sense to me

File: config/routes.rb (hunk 4 above)
Comment (KirbyDD) [reply config/routes.rb:28]: well this part isnt the API yet. this is just the routing to send a user to github to authorize use to use their info

File: app/assets/stylesheets/application.sass (hunk 3 above)
Comment (KirbyDD) [reply app/assets/stylesheets/application.sass:7]: yep just so i could style the link as a button

**Pull request merged**